cmd/petstore/main.go

```diff
@@ -1,4 +1,4 @@
-package main
+package main_test
```

Comment:
not a criticism, but why did you opt to separate these tests from the main package?

Reply:
The `xxxx_test` format was opted for in order to allow for more comprehensive testing with indirectly causing import cycles. The only time I deviate from this format is where private functions NEED to be tested. Ideally private functions should be tested through their public counterparts, and thats the trajectory I want to move in with this.